src/main/java/org/kainos/ea/resources/WebService.java

```diff
@@ -84,4 +89,27 @@ public Response getJobSpecification( @PathParam("id") int id ) throws SQLExcepti
         }
     }
 
+    @POST
+    @Path("/register")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response registerUser(UserRequest user) {
+        if (userValidator.isValidUser(user)) {
+            try {
+                int id = userService.registerUser(user);
+                return Response.status(HttpStatus.CREATED_201).entity(id).build();
+            } catch (DatabaseConnectionException | SQLException e) {
+                System.out.println(e);
+                return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+            }
+            catch (DataNotFoundException e) {
```

Comment:
I don't see this exception being thrown anywhere, can you throw it when the roleID passed in does not exist in the db?

Reply:
exception thrown in UserService